packages/kbn-test/src/mocha/junit_report_generation.js

```diff
@@ -136,7 +136,10 @@ export function setupJUnitReportGeneration(runner, options = {}) {
       }
     });
 
-    const reportPath = resolve(rootDirectory, `target/junit/TEST-${reportName}.xml`);
+    const reportPath = resolve(
+      rootDirectory,
+      `target/junit/TEST-${process.env.JOB ? process.env.JOB + '-' : ''}${reportName}.xml`
```

Comment:
Same thing, can we make `process.env.JOB` a directory and default to `.` if it's not defined?